packages/ember-glimmer/tests/integration/helpers/unbound-test.js

```diff
@@ -0,0 +1,545 @@
+
+    this.runTask(() => {
+      setProperties(this.context, {
+        prefix: 'before',
+        value: 'core',
+        suffix: 'after'
+      });
+    });
+
+    this.assertText('before-core-bar before-core-bar bar-core-after bar-core-after');
+  }
+
+  ['@htmlbars should be able to render unbound forms of multi-arg helpers']() {
+    this.registerHelper('-fauxconcat', (params) => params.join(''));
+
+    this.render(`{{-fauxconcat foo bar bing}} {{unbound (-fauxconcat foo bar bing)}}`, {
+      foo: 'a',
+      bar: 'b',
+      bing: 'c'
+    });
+
+    this.assertText('abc abc');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('abc abc');
+
+    this.runTask(() => set(this.context, 'bar', 'X'));
+
+    this.assertText('aXc abc');
+
+    this.runTask(() => set(this.context, 'bar', 'b'));
+
+    this.assertText('abc abc');
+  }
+
+  ['@htmlbars should be able to render an unbound helper invocation for helpers with dependent keys']() {
+    this.registerHelper('-capitalizeName', {
+      destroy() {
+        this.removeObserver('value.firstName');
+        this._super(...arguments);
+      },
+
+      compute([value]) {
+        if (this.get('value')) {
+          this.removeObserver('value.firstName');
+        }
+        this.set('value', value);
+        this.addObserver('value.firstName', this, this.recompute);
+        return (value ? get(value, 'firstName').toUpperCase() : '');
+      }
+    });
+
+    this.registerHelper('-concatNames', {
+      destroy() {
+        this.teardown();
+        this._super(...arguments);
+      },
+      teardown() {
+        this.removeObserver('value.firstName');
+        this.removeObserver('value.lastName');
+      },
+      compute([value]) {
+        if (this.get('value')) {
+          this.teardown();
+        }
+        this.set('value', value);
+        this.addObserver('value.firstName', this, this.recompute);
+        this.addObserver('value.lastName', this, this.recompute);
+        return (value ? get(value, 'firstName') : '') + (value ? get(value, 'lastName') : '');
+      }
+    });
+
+    this.render(`{{-capitalizeName person}} {{unbound (-capitalizeName person)}} {{-concatNames person}} {{unbound (-concatNames person)}}`, {
+      person: EmberObject.create({
+        firstName: 'shooby',
+        lastName:  'taylor'
+      })
+    });
+
+    this.assertText('SHOOBY SHOOBY shoobytaylor shoobytaylor');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('SHOOBY SHOOBY shoobytaylor shoobytaylor');
+
+    this.runTask(() => set(this.context, 'person.firstName', 'sally'));
+
+    this.assertText('SALLY SHOOBY sallytaylor shoobytaylor');
+
+    this.runTask(() => set(this.context, 'person', EmberObject.create({
+      firstName: 'shooby',
+      lastName:  'taylor'
+    })));
+
+    this.assertText('SHOOBY SHOOBY shoobytaylor shoobytaylor');
+  }
+
+  ['@htmlbars should be able to render an unbound helper invocation in #each helper']() {
```

Comment:
Port of https://github.com/emberjs/ember.js/blob/bf067f7370c3c3381352d68ef5dfe55d1178c6f7/packages/ember-htmlbars/tests/helpers/unbound_test.js#L407-L431